src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -117,30 +119,191 @@ def get_shares(self) -> List[torch.Tensor]:
         """
         return self.shares
 
-    def add(self, y):
-        """Apply the "add" operation between "self" and "y".
+    @staticmethod
+    def sanity_checks(
+        x: "ReplicatedSharedTensor",
+        y: Union[int, float, torch.Tensor, "ReplicatedSharedTensor"],
+        op_str: str,
+    ) -> "ReplicatedSharedTensor":
+        """Check the type of "y" and covert it to share if necessary.
+
+        Args:
+            x (ReplicatedSharedTensor): Typically "self".
+            y (Union[int, float, torch.Tensor, "ReplicatedSharedTensor"]): Tensor to check.
+            op_str (str): String operator.
+
+        Returns:
+            ReplicatedSharedTensor: the converted y value.
+
+        Raises:
+            ValueError: if both values are shares and they have different uuids
+            ValueError: if both values have different number of shares.
+        """
+        if not isinstance(y, ReplicatedSharedTensor):
+            if x.session_uuid is not None:
+                session = sympc.session.get_session(str(x.session_uuid))
+                ring_size = session.ring_size
+                config = session.config
+            else:
+                ring_size = x.ring_size
+                config = x.config
+
+            y = ReplicatedSharedTensor(shares=[y], ring_size=ring_size, config=config)
+
+        elif y.session_uuid and x.session_uuid and y.session_uuid != x.session_uuid:
+            raise ValueError(
+                f"Session UUIDS did not match {x.session_uuid} {y.session_uuid}"
+            )
+        elif len(x.shares) != len(y.shares):
+            raise ValueError("Both RSTensors should have equal number of shares.")
+
+        return y
+
+    def __apply_public_op(
+        self, y: Union[torch.Tensor, float, int], op_str: str
+    ) -> "ReplicatedSharedTensor":
+        """Apply an operation on "self" which is a RSTensor and a public value.
+
+        Args:
+            y (Union[torch.Tensor, float, int]): Tensor to apply the operation.
+            op_str (str): The operation.
+
+        Returns:
+            ReplicatedSharedTensor. The operation "op_str" applied on "self" and "y"
+
+        Raises:
+            ValueError: If "op_str" is not supported.
+        """
+        rank = 0
+        nr_parties: int = 1
+        session_uuid = self.session_uuid
+        if session_uuid is not None:
+            session = sympc.session.get_session(str(self.session_uuid))
+            ring_size = session.ring_size
+            config = session.config
+            rank = session.rank
+            nr_parties = session.nr_parties
+        else:
+            ring_size = self.ring_size
+            config = self.config
+
+        op = getattr(operator, op_str)
+        shares = self.shares
+        if op_str in {"add", "sub"}:
+            if rank - 1 != 0:
```

Comment:
Since in RSTensor `nshares=nparties-1` ,second party (rank=1) would not have the first share.Since Linear operations add constant to first shares.I could have done it as `rank!=1`,it seemed like hard coding it.

Reply:
Q: Why not directly check `if rank != 1`?

Reply:
Sure ,George,I would change it , it seemed like hard coding the values,as we had made an assumption on the shares.